### ISSUE TITLE
Public shared mind maps are no longer openable - download only

### ISSUE THREAD

**Issue by Mer0me** [ACTom/files_mindmap#187]
Since v0.0.29 mind maps shared links cannot be opened, it only proposes to download the .km
Files are well opened in nextcloud with connected accounts.
Share the parent folder instead of the map file doesn't work.

**Comment by Mer0me** [comment]
I confirm that going back to 0.0.28 solves the problem.

**Comment by ACTom** [comment]
@smarinier  I don't have time to fix it these days, can you help solve this problem?

**Comment by smarinier** [comment]
> @smarinier I don't have time to fix it these days, can you help solve this problem?

Yes. But @Mer0me can you be more precise in your issue description (screen shots ?). Min Map Shared Links can be many things for me.

**Comment by Mer0me** [comment]
Sure, if I can help.

This map is shared by link : 
![image](https://github.com/ACTom/files_mindmap/assets/6248852/5867dd1d-dc32-4446-9833-6718f7d87de5)


I can open it, as owner, into Nextcloud : 
![image](https://github.com/ACTom/files_mindmap/assets/6248852/e3a363da-b788-40a3-8536-6f8c4e484b57)

But the public link, does not work :
![image](https://github.com/ACTom/files_mindmap/assets/6248852/440a13a6-d19e-4965-95c9-817e00fe9a9d)

I only can download the .km
With previous version of the app, external sharees was able to see the map in read only mode.

**Comment by Sicoval31** [comment]
Same problem.

**Comment by smarinier** [comment]
Could you confirm the downgrade of files_mindmap to 0.28 ? I mean : keep exactly the same NC, but change only apps/files_mindmap folder. Because in my case, if I'm doing this action, I got the same bug with 0.28. Are you sure this is not  due to another upgrade ?

**Comment by Mer0me** [comment]
I confirm : 
Nextcloud v27.1.1
Mind Map 0.0.28
I can open public link on shared mind map.

To downgrade, I've use this method : 
1. Disable the mind map app in nextcloud
2. Download v0.0.28 here : https://github.com/ACTom/files_mindmap/releases/download/v0.0.28/files_mindmap-0.0.28.tar.gz
3. Replace nextcloud/apps/files_mindmap folder by the uncompressed folder of v0.0.28 downloaded
4. Enable the mind map app in nextcloud

**Comment by smarinier** [comment]
I confirm the bug is in 0.28 too.

It depends the way of doing it. If you go through files, then shared, it works. If your going directly to index.php/apps/files/shareoverview, it doesn't work. In fact, this is quite random, as it depend on the order of some initializing actions.

As I'm on it, i will fix it. But it is clearly not caused by 0.29 version.

As far i can see, this is due to two things (mindmap.js):

1)  $(document).ready(function(){
   -> this is an old fashion way, all other plugins use now
window.addEventListener('DOMContentLoaded', function () {
 
 The order of actions depends then on things I don't really want to investigate in. ;). (what i called "quite random")
 This seem also (but it's an hypothesis) "play" with the chance that "files_sharing" module is after "files_mindmap". This is not a good thing to rely on.
 
2) The SharingFileList use a copy of the main FileList, and it's done "at a certain time".

We need to use the "recommended way" : 
   OC.Plugins.register('OCA.Files.FileList'
That indicates to a fileList that it must call attached plugins on initialisation.

I will submit a fix proposal soon.

**Comment by smarinier** [comment]
Pull request pushed. @ACTom could you take a look ?

**Comment by ACTom** [comment]
@smarinier I tested it and found that shared links can still only be downloaded, regardless of whether it's a shared directory or a shared file, in this case a link share.

**Comment by smarinier** [comment]
Hi @ACTom , could you precise the use case. Because when i test it works fine for me.

**Comment by ACTom** [comment]
@smarinier https://cloud.libreofficechina.org/s/NONM2pxFknUn86K https://cloud.libreofficechina.org/s/pndz3ZfeHZaCSzC

**Comment by smarinier** [comment]
PR #189 sumbitted

**Comment by Mer0me** [comment]
Some news on that ?

**Comment by smarinier** [comment]
I've submitted a fix with a PR on 30 october. It's up to @ACTom to validate it i guess. You can also test it, if you want.

**Comment by ACTom** [comment]
Merged.
Release a new version [(0.0.30)](https://github.com/ACTom/files_mindmap/releases/tag/v0.0.30) to fix it.
Thanks @smarinier .
